### PR TITLE
Add CLI skeleton

### PR DESCRIPTION
**Pull request by LucianPy**
Add initial CLI functionality for creating and executing commands as well as accessing their attributes
Cut -tool from module name, commands will now be of format elastic-agent-changelog [command]

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,3 +1,10 @@
-module github.com/elastic/elastic-agent-changelog-tool
+module github.com/elastic/elastic-agent-changelog
```
Comment (endorama) [inline go.mod:1]: May you help me understand why this change has been introduced? Given how imports work in go this is very confusing.

If we want a nicer name I would prefer renaming the repository.

File: internal/cobraext/command.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+type CommandContext string
+
+const (
+	// ContextGlobal means the command runs in a global context, agnostic of any
+	// specific package.
+	ContextGlobal CommandContext = "global"
+
+	// ContextPackage means the command runs in the contexts of a specific package.
+	ContextPackage CommandContext = "package"
+)
```
Comment (endorama) [inline internal/cobraext/command.go:24]: What is the purpose of `CommandContext` in this CLI?

File: cmd/root.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+// or more contributor license agreements. Licensed under the Elastic License
+// 2.0; you may not use this file except in compliance with the Elastic License
+// 2.0.
+
+package cmd
+
+import (
+	"sort"
+
+	"github.com/elastic/elastic-agent-changelog/internal/cobraext"
```
Comment (endorama) [inline cmd/root.go:11]: Unless we plan to use `cobraext` I would not include it at this stage, as it doesn't seem to be used at the moment. We can introduce this later if needed.

File: cmd/root.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+// RootCmd creates and returns root cmd for elastic-agent-changelog.
+func RootCmd() *cobra.Command {
+	rootCmd := &cobra.Command{
+		Use:          "elastic-agent-changelog",
+		Short:        "elastic-agent-changelog - Command line tool used for managing the change log for Elastic Agent and related components, including Beats.",
+		SilenceUsage: true,
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			return cobraext.ComposeCommandActions(cmd, args)
+		},
```
Comment (endorama) [inline cmd/root.go:26]: This is a `noop` so please either remove it or use it.

File: go.mod (hunk 1 above)
Comment (LucianPy) [reply go.mod:1]: This was originally done due to the way of how installing go packages work in the context of running a cli. Commands will take the module name by default, so elastic-agent-changelog [command] was the specs one, -tool just didn't respect the command format needed.

File: internal/cobraext/command.go (hunk 2 above)
Comment (LucianPy) [reply internal/cobraext/command.go:24]: This might be useful in the future for providing the context when running in scripts etc. But you are correct, no need to include it at this point.

File: cmd/root.go (hunk 3 above)
Comment (LucianPy) [reply cmd/root.go:11]: This is used in root, it provides access to the command structured, along with other functionality.

Comment (LucianPy) [reply cmd/root.go:11]: It will also be needed for future improvements in root.go

File: cmd/root.go (hunk 4 above)
Comment (LucianPy) [reply cmd/root.go:26]: Correct, will remove it.

File: go.mod (hunk 1 above)
Comment (endorama) [reply go.mod:1]: :+1: I'll check with him on this

File: cmd/root.go (hunk 3 above)
Comment (endorama) [reply cmd/root.go:11]: To my understanding `cobraext` main functionality is wrapping a standard `cobra.Command` to add `CommandContext`. I'm not sure is worth the abstraction if we don't plan to add contexts to commands. We are not going to use `ComposeCommandActions` (as discussed in https://github.com/elastic/elastic-agent-changelog-tool/pull/2#discussion_r799400217) so `cobraext` does not provide useful functionality at the moment.

Based on
> It will also be needed for future improvements in root.go
you may already think of improvements that would make sense to bring here.

File: go.mod (hunk 1 above)
Comment (endorama) [reply go.mod:1]: Let's rename the module to `github.com/elastic/elastic-agent-changelog-tool`
